service/view_building_coordinator.cc

```diff
@@ -184,18 +185,25 @@ future<std::optional<std::pair<group0_guard, view_building_coordinator::vbc_stat
 
     // Since vbc_state is immutable, select new base table to process only if there are no other changes to the state.
     if (!state.currently_processed_base_table && !state.tasks.empty() && cmuts.empty()) {
+        SCYLLA_ASSERT(state.targets_with_staging_sstables.empty());
+        
         auto& base_id = state.tasks.cbegin()->first;
         vbc_logger.info("Start building views for base table: {}", base_id);
 
         auto mut = co_await _sys_ks.make_vbc_processing_base_mutation(guard.write_timestamp(), base_id);
         cmuts.emplace_back(std::move(mut));
+        notify_others_to_detect_staging = true;
```

Comment:
Why do we decide to nudge other nodes to detect staging sstables when starting build of a new view? Shouldn't we do it after we finish building the current view?

Reply:
outdated